content/drone-plugins/drone-s3-cache/index.md

```diff
@@ -0,0 +1,174 @@
+---
+date: 2017-08-29T00:00:00+00:00
+title: Amazon S3 Cache
+author: drone-plugins
+tags: [ amazon, aws, s3, cache ]
+repo: drone-plugins/drone-s3-cache
+logo: amazon_s3.svg
+image: plugins/s3-cache
+---
+
+The S3 cache plugin can be used to preserve files and directories between builds. The below pipeline configuration demonstrates simple usage:
+
+```yaml
+pipeline:
+  restore-cache:
+    image: plugins/s3-cache:1
+    pull: true
+    url: http://minio.company.com
+    access_key: myaccesskey
+    secret_key: supersecretKey
+    restore: true
+
+  build:
+    image: node
+    commands:
+      - npm install
+
+  rebuild-cache:
+    image: plugins/s3-cache:1
+    pull: true
+    url: http://minio.company.com
+    access_key: myaccesskey
+    secret_key: supersecretKey
+    rebuild: true
+    mount:
+      - node_modules
+
+  flush_cache:
+    image: plugins/s3-cache:1
+    pull: true
+    url: http://minio.company.com
+    access_key: myaccesskey
+    secret_key: supersecretKey
+    flush: true
+    flush_age: 14
+```
+
+Use additional custom cache directories:
+
+```diff
+pipeline:
+  restore-cache:
+    image: plugins/s3-cache:1
+    pull: true
+    url: http://minio.company.com
+    access_key: myaccesskey
+    secret_key: supersecretKey
+    restore: true
+
+  build:
+    image: node
+    commands:
+      - npm install
+
+  rebuild-cache:
+    image: plugins/s3-cache:1
+    pull: true
+    url: http://minio.company.com
+    access_key: myaccesskey
+    secret_key: supersecretKey
+    rebuild: true
+    mount:
+-     - node_modules
++     - <yourstuffhere>
++     - <morestuffhere>
+
+  flush_cache:
+    image: plugins/s3-cache:1
+    pull: true
+    url: http://minio.company.com
+    access_key: myaccesskey
+    secret_key: supersecretKey
+    flush: true
+    flush_age: 14
+```
+
+Example configuration using credentials from secrets:
+
+```diff
+pipeline:
+  restore-cache:
+    image: plugins/s3-cache:1
+    pull: true
+    url: http://minio.company.com
+-   access_key: myaccesskey
+-   secret_key: supersecretKey
++   secrets: [ cache_s3_access_key, cache_s3_secret_key ]
+    restore: true
+
+  build:
+    image: node
+    commands:
+      - npm install
+
+  rebuild-cache:
+    image: plugins/s3-cache:1
+    pull: true
+    url: http://minio.company.com
+-   access_key: myaccesskey
+-   secret_key: supersecretKey
++   secrets: [ cache_s3_access_key, cache_s3_secret_key ]
+    rebuild: true
+    mount:
+      - node_modules
+
+  flush_cache:
+    image: plugins/s3-cache:1
+    pull: true
+    url: http://minio.company.com
+-   access_key: myaccesskey
+-   secret_key: supersecretKey
++   secrets: [ cache_s3_access_key, cache_s3_secret_key ]
```

Comment:
When possible, we should use well known secret names. In this case we should be using `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` values.

Reply:
But the plugin does not use the well known variable names yet

Reply:
I would recommend the plugin updated to accept AWS_SECRET_ variables